### ISSUE TITLE
[.pylint] % formatting in logging

### ISSUE THREAD

**Issue by lachmanfrantisek** [user-cont/kwaciaren#9]
We have many occurrences of this:

```
Use % formatting in logging functions and pass the % parameters as arguments
```

We are mostly using new `.format` syntax -- is this optimalisation really necessary?

---

PyLint_W1202 

https://stackoverflow.com/questions/34619790/pylint-message-logging-format-interpolation

**Comment by TomasTomecek** [comment]
I personally prefer using `logger.debug("hi there %s", the_dude)` over `.format(`. Hence for me the warning is useful. It's really up to you, we can vote.

**Comment by jpopelka** [comment]
Yeah, I constantly poke people in comments about it, but some people can't even be consistent and chose one way how to format strings because they give a f* about warnings.

So let's be pragmatic for a moment:
The only way (I see) why using `%` over `format()` (or better `f-strings`) is that the string doesn't need to be formatted if it's in `debug()` and you run the code with `info` or `warn`.
With nowadays computing power, I don't think that difference is even perceptible.
On the other hand, using `f-strings` over `%` makes reading others' code sooo much easier that I'm inclined to consider this warning obsolete.

**Comment by TomasTomecek** [comment]
@jpopelka please, teach us f-strings then :)

**Comment by jpopelka** [comment]
> please, teach us f-strings then :)

not sure how to react, since I know all (but one) of us know it.

What about ?
```python
string = f"Hello {some_variable}. You are {another_variable}."
```
is there anything else you need to know ?

**Comment by TomasTomecek** [comment]
Actually half of this: https://pyformat.info/

I always forget how to do `%r` with `.format` and have no idea how to do that with f-strngs. Also padding and floats are used so rarely that I keep forgetting those constantly.

Some time ago I started this cheatsheet: https://github.com/TomasTomecek/mysterious-keybinds#python Having a clear table which all of us could print and stuff behind our monitors would be mind-blowing. Probably even nobel price-winning.
